src/modules/Servicehosting/Service.php

```
@@ -1155,59 +1128,61 @@ public function getHpSearchQuery($data): array
             throw new InformationException('Cannot remove hosting plan which has active accounts');
         }
-        $this->di['db']->trash($model);
+        $this->di['em']->remove($model);
+        $this->di['em']->flush();
         $this->di['logger']->info('Deleted hosting plan %s', $id);
 
         return true;
     }
 
-    public function toHostingHpApiArray(\Model_ServiceHostingHp $model, $deep = false, $identity = null): array
+    public function toHostingHpApiArray(ServiceHostingHp $model, $deep = false, $identity = null): array
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$deep'. (undefined)

(UnusedFormalParameter)